kubeletclient/kubeletclient.go

```diff
@@ -0,0 +1,113 @@
+package kubeletclient
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/intel/multus-cni/checkpoint"
+	"github.com/intel/multus-cni/logging"
+	"github.com/intel/multus-cni/types"
+	"golang.org/x/net/context"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/kubelet/apis/podresources"
+	podresourcesapi "k8s.io/kubernetes/pkg/kubelet/apis/podresources/v1alpha1"
+	"k8s.io/kubernetes/pkg/kubelet/util"
+)
+
+const (
+	defaultKubeletSocketFile   = "kubelet.sock"
+	defaultPodResourcesPath    = "/var/lib/kubelet/pod-resources"
+	defaultPodResourcesMaxSize = 1024 * 1024 * 16 // 16 Mb
+)
+
+var (
+	kubeletSocket string
+)
+
+// GetResourceClient returns an instance of ResourceClient interface initialized with Pod resource information
+func GetResourceClient() (types.ResourceClient, error) {
+	// If Kubelet resource API endpoint exist use that by default
+	// Or else fallback with checkpoint file
+	if hasKubeletAPIEndpoint() {
+		logging.Printf(logging.VerboseLevel, "GetResourceClient(): using Kubelet resource API endpoint")
+		return getKubeletClient()
+	} else {
+		logging.Printf(logging.VerboseLevel, "GetResourceClient(): using Kubelet device plugin checkpoint")
+		return checkpoint.GetCheckpoint()
+	}
+}
+
+func getKubeletClient() (types.ResourceClient, error) {
+	newClient := &kubeletClient{}
+	if kubeletSocket == "" {
```

Comment:
This will always be false as ``kubeletSocket`` is set in ``hasKubeletAPIEndpoint``.

Reply:
> This will always be false as `kubeletSocket` is set in `hasKubeletAPIEndpoint`.

Good catch! I've corrected the variable in `hasKubeletAPIEndpoint` which should be only local.